### PR TITLE
Msglib fast dds

### PR DESCRIPTION
**Pull request by hongdami**
Issue Number: 

Objective of pull request:

## Pull request checklist

Your PR fulfills the following requirements:
- [ ] [Issue](https://github.com/lava-nc/lava/issues) created that explains the change and why it's needed
- [ ] Tests are part of the PR (for bug fixes / features)
- [ ] [Docs](https://github.com/lava-nc/docs) reviewed and added / updated if needed (for bug fixes / features)
- [ ] PR conforms to [Coding Conventions](https://lava-nc.org/developer_guide.html#coding-conventions)
- [ ] [PR applys BSD 3-clause or LGPL2.1+ Licenses](https://lava-nc.org/developer_guide.html#add-a-license) to all code files
- [ ] Lint (`flakeheaven lint src/lava tests/`) and (`bandit -r src/lava/.`) pass locally
- [ ] Build tests (`pytest`) passes locally

## Pull request type

 

 

Please check your PR type:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe): 

## What is the current behavior?

-

## What is the new behavior?

-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/dds.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  DDSSubscriberPtr dds_subscriber_ = nullptr;
+};
+
+using DDSPtr = std::shared_ptr<DDS>;
+
+class DDSManager {
+ public:
+  ~DDSManager();
+  DDSPtr AllocDDS(const size_t &size,
+                       const size_t &nbytes,
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/dds.h:49]: code style

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/dds_channel.cc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+namespace message_infrastructure {
+
+DDSChannel::DDSChannel(const size_t &depth,
+                       const size_t &nbytes,
+                       const std::string &topic_name,
+                       const DDSTransportType &dds_transfer_type) {
+  printf("DDSChannel: get dds\n");
+  dds_ = GetDDSManager().AllocDDS(depth, nbytes, topic_name, dds_transfer_type);
+  printf("DDSChannel: set port\n");
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/dds_channel.cc:18]: replace by log func

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/dds.cc [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+DDSManager::~DDSManager() {
+  DeleteAllDDS();
+}
+
+DDS::DDS(const size_t &max_samples,
+         const size_t &nbytes,
+         const std::string &topic_name,
+         const DDSTransportType &dds_transfer_type) {
+#if defined(FASTDDS)
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/dds.cc:42]: how to handle CycloneDDS?
pls add a enum of DDS BACKEND as input parameter.

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/protos/fast_dds/metadata.h [hunk 4]
```diff
@@ -0,0 +1,214 @@
+// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/protos/fast_dds/metadata.h:1]: these files are generated from metadata.idl/metadata.xml?

**Review by killight98 — CHANGES_REQUESTED** [review]

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/protos/fast_dds/metadata.h (hunk 4 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/protos/fast_dds/metadata.h:1]: yes, created by Fast-DDS-Gen tool

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/dds.cc (hunk 3 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/dds.cc:44]: I have update the DDS Backend type

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/dds_channel.cc (hunk 2 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/dds_channel.cc]: sure

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/dds.h (hunk 1 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/dds.h]: updated, fixing install error ...

**Comment by killight98** [comment]
@hongdami TODO list:

- [ ] Support TCP & UDP transport type.
- [ ] Use `metadata.xml` with specific parameters.

File: src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt [hunk 5]
```diff
@@ -12,10 +12,12 @@ include_directories(../message_infrastructure/csrc)
 
 add_executable(
   test_messaging_infrastructure
-  cpptest/test_multiprocessing.cc
+  # cpptest/test_multiprocessing.cc
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt:15]: restore this ut

File: src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt [hunk 6]
```diff
@@ -12,10 +12,12 @@ include_directories(../message_infrastructure/csrc)
...
   # cpptest/test_channel.cc
+  # cpptest/test_shm_delivery.cc
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt:17]: remove this line

File: src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+  MetaDataPtr metadata = std::make_shared<MetaData>();
+  metadata->nd = 1;
+  metadata->type = 7;
+  metadata->elsize = 8;
+  metadata->total_size = 1;
+  metadata->dims[0] = 1;
+  metadata->strides[0] = 1;
+  metadata->mdata =
+    (reinterpret_cast<char*>
+    (malloc(sizeof(int64_t)+sizeof(MetaData)))+sizeof(MetaData));
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc:106]: pls remove offset value (sizeof(MetaData)) at this ut.

**Review by killight98 — CHANGES_REQUESTED** [review]

File: src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+void dds_target_fn_a2_bound(int loop,
+                        AbstractChannelPtr a1_to_a2,
+                        AbstractChannelPtr a2_to_a1,
+                        AbstractActor* actor_ptr) {
+  actor_ptr->SetStopFn(dds_stop_fn);
+  auto from_a1 = a1_to_a2->GetRecvPort();
+  from_a1->Start();
+  auto to_a1   = a2_to_a1->GetSendPort();
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc:56]: code style

File: src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+  to_a1->Start();
+  while ((loop--)&&!actor_ptr->GetStatus()) {
+    MetaDataPtr data = from_a1->Recv();
+    (*reinterpret_cast<int64_t*>(data->mdata))++;
+    to_a1->Send(data);
+    // free(reinterpret_cast<char*>(data->mdata)-sizeof(MetaData));
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc:62]: restore free func

**Review by killight98 — CHANGES_REQUESTED** [review]

File: src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt (hunk 5 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/test/CMakeLists.txt]: ok

File: src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc (hunk 7 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc]: malloc(sizeof(int64_t)

File: src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc (hunk 8 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc]: fix line 56

File: src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc (hunk 9 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc]: free(data->mdata)

File: src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+    (reinterpret_cast<char*>
+    (malloc(sizeof(int64_t)+sizeof(MetaData)))+sizeof(MetaData));
+  *reinterpret_cast<int64_t*>(metadata->mdata) = 1;
+
+  MetaDataPtr mptr;
+  int loop = 100000;
+  int i = 0;
+  while (loop--) {
+    if (!(loop % 1000))
+      printf("At iteration : %d * 1000\n", i++);
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc:147]: replace by LAVA_DUMP

File: src/lava/magma/runtime/message_infrastructure/test/test_channel.py [hunk 11]
```diff
@@ -200,6 +201,38 @@ def test_grpcchannel(self):
...
+        builder2 = Builder()
+        mp.build_actor(recv_port_fn, builder1)
+        mp.build_actor(send_port_fn, builder2)
+
+        time.sleep(0.1)
+        mp.stop(True)
+
 
 if __name__ == "__main__":
+    print(SupportDDSChannel)
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/test_channel.py:237]: remove this print

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/core/channel_factory.h [hunk 12]
```diff
@@ -28,7 +32,19 @@ class ChannelFactory {
                                 const std::string &src_name,
                                 const std::string &dst_name);
 
-  friend ChannelFactory& GetChannelFactory();
+#if defined(DDS_CHANNEL)
+  AbstractChannelPtr GetDefDDSChannel(const int &dds_depth,
+                                   const size_t &nbytes,
+                                   const std::string &topic_name,
+                                   const DDSTransportType &transport_type,
+                                   const DDSBackendType &dds_backend) {
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/core/channel_factory.h:40]: - remove nbytes and calculate it by metadata value.
- rename dds_depth as size, and keep `max_samples` at dds api
- change parameter order as following:
```
AbstractChannelPtr GetDefDDSChannel(
                                   const std::string &topic_name,
                                   const DDSTransportType &transport_type,
                                   const DDSBackendType &dds_backend,
                                   const int &size)
```

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/dds.h [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+
+using DDSSubscriberPtr = std::shared_ptr<DDSSubscriber>;
+
+class DDS {
+ public:
+  DDS(const size_t &max_samples,
+      const size_t &nbytes,
+      const std::string &topic_name,
+      const DDSTransportType &dds_transfer_type,
+      const DDSBackendType &dds_backend);
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/dds.h:42]: change order and put max_samples as last one.

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/dds.h [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+using DDSPtr = std::shared_ptr<DDS>;
+
+class DDSManager {
+ public:
+  ~DDSManager();
+  DDSPtr AllocDDS(const size_t &size,
+                  const size_t &nbytes,
+                  const std::string &topic_name,
+                  const DDSTransportType &dds_transfer_type,
+                  const DDSBackendType &dds_backend);
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/dds.h:66]: same with DDS

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc [hunk 15]
```diff
@@ -0,0 +1,363 @@
...
+  stop_ = false;
+  return 0;
+}
+
+void FastDDSPublisher::InitDataWriter() {
+  DataWriterQos wqos;
+  wqos.history().kind = KEEP_LAST_HISTORY_QOS;
+  wqos.history().depth = max_samples_;
+  wqos.resource_limits().max_samples = max_samples_;
+  wqos.resource_limits().allocated_samples = max_samples_/2;
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc:67]: code style

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc [hunk 16]
```diff
@@ -0,0 +1,363 @@
...
+  wqos.reliable_writer_qos().times.heartbeatPeriod.seconds = 2;
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc:68]: define macro for this value

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc [hunk 17]
```diff
@@ -0,0 +1,363 @@
...
+  wqos.reliable_writer_qos().times.heartbeatPeriod.nanosec = 200 * 1000 * 1000;
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc:69]: define macro for this value

**Review by killight98 — CHANGES_REQUESTED** [review]

File: src/lava/magma/runtime/message_infrastructure/CMakeLists.txt [hunk 18]
```diff
@@ -69,13 +68,48 @@ if(GRPC_CHANNEL)
...
+    add_definitions(-DFASTDDS)
+    set(FASTDDS_SRC
+      "message_infrastructure/csrc/channel/dds/fast_dds.cc"
+      "message_infrastructure/csrc/channel/dds/protos/fast_dds/metadata.cc"
+      "message_infrastructure/csrc/channel/dds/protos/fast_dds/metadataPubSubTypes.cc"
+      ${DDS_CHANNEL_SRCS})
+  endif()
+
+  if(FASTDDS_ENABLE OR CycloneDDS_ENABLE)
+    add_definitions(-DDDS_CHANNEL)
```
Comment (hexu33) [inline src/lava/magma/runtime/message_infrastructure/CMakeLists.txt:100]: Seems there is no need to judge FASTDDS_ENABLE or CycloneDDS_ENABLE to add DDS_CHANNEL definition.

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/message_infrastructure_py_wrapper.cc [hunk 19]
```diff
@@ -105,6 +109,36 @@ PYBIND11_MODULE(MessageInfrastructurePywrapper, m) {
     return false;
 #endif
   });
+
+  py::enum_<DDSTransportType> (m, "DDSTransportType")
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/message_infrastructure_py_wrapper.cc:113]: put `DDSTransportType` and `DDSBackendType` under `defined(DDS_CHANNEL)`

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/__init__.py [hunk 20]
```diff
@@ -11,10 +11,16 @@
 from MessageInfrastructurePywrapper import RecvPort
 from MessageInfrastructurePywrapper import AbstractTransferPort
 from MessageInfrastructurePywrapper import support_grpc_channel
+from MessageInfrastructurePywrapper import DDSTransportType
+from MessageInfrastructurePywrapper import DDSBackendType
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/__init__.py:15]: if SupportDDSChannel, import DDSTransportType, DDSBackendType

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/core/utils.h [hunk 21]
```diff
@@ -31,6 +31,29 @@ enum ChannelType {
   SOCKETCHANNEL = 3
 };
 
+enum DDSTransportType {
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/core/utils.h:34]: i think we need to put following enums under `channel/dds`.

File: src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc [hunk 22]
```diff
@@ -0,0 +1,158 @@
...
+  auto to_mp   = a1_to_mp->GetSendPort();
+  to_mp->Start();
+  auto to_a2   = a1_to_a2->GetSendPort();
+  to_a2->Start();
+  auto from_a2 = a2_to_a1->GetRecvPort();
+  from_a2->Start();
+  while ((loop--)&&!actor_ptr->GetStatus()) {
+    MetaDataPtr data = from_mp->Recv();
+    (*reinterpret_cast<int64_t*>(data->mdata))++;
+    to_a2->Send(data);
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc:35]: after send, need to free mdata.

File: src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc [hunk 23]
```diff
@@ -0,0 +1,158 @@
...
+    data = from_a2->Recv();
+    (*reinterpret_cast<int64_t*>(data->mdata))++;
+    to_mp->Send(data);
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc:38]: after send, need to free mdata.

**Review by killight98 — CHANGES_REQUESTED** [review]

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc [hunk 24]
```diff
@@ -0,0 +1,363 @@
...
+  }
+  stop_ = true;
+}
+
+void FastDDSPubListener::on_publication_matched(
+        eprosima::fastdds::dds::DataWriter*,
+        const eprosima::fastdds::dds::PublicationMatchedStatus& info) {
+  if (info.current_count_change == 1) {
+    matched_++;
+    first_connected_ = true;
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc:162]: first_connected_ is redundant?

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc [hunk 25]
```diff
@@ -0,0 +1,363 @@
...
+      LAVA_DEBUG(LOG_DDS, "Publish a data\n");
+    }
+    return true;
+  }
+  return false;
+}
+
+void FastDDSPublisher::Stop() {
+  LAVA_LOG(LOG_DDS, "Stop FastDDS Publisher, waiting unmatched...\n");
+  while (listener_->matched_ > 0) {
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc:138]: if others are ROS2 node, must to wait matched node to zero at FastDDSPublisher?

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.h [hunk 26]
```diff
@@ -0,0 +1,146 @@
...
+                   type_(new DDSMetaDataPubSubType()),
+                   stop_(true),
+                   max_samples_(max_samples),
+                   nbytes_(nbytes),
+                   topic_name_(topic_name),
+                   dds_transfer_type_(dds_transfer_type)
+                    {};
+  ~FastDDSSubscriber() override;
+  int Init();
+  void Run();
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.h:121]: empty func?

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.h [hunk 27]
```diff
@@ -0,0 +1,146 @@
...
+
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/publisher/Publisher.hpp>
+#include <fastdds/dds/publisher/DataWriter.hpp>
+#include <fastdds/dds/publisher/DataWriterListener.hpp>
+#include <fastdds/dds/topic/Topic.hpp>
+#include <fastdds/dds/topic/TypeSupport.hpp>
+#include <fastdds/dds/subscriber/Subscriber.hpp>
+#include <fastdds/dds/subscriber/DataReader.hpp>
+#include <fastdds/dds/subscriber/SampleInfo.hpp>
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.h:25]: include order

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc [hunk 28]
```diff
@@ -0,0 +1,363 @@
...
+
+#include <fastdds/rtps/transport/shared_mem/SharedMemTransportDescriptor.h>
+#include <fastdds/rtps/transport/TCPv4TransportDescriptor.h>
+#include <fastdds/rtps/transport/TCPv6TransportDescriptor.h>
+#include <fastdds/rtps/transport/UDPv4TransportDescriptor.h>
+#include <fastdds/rtps/transport/UDPv6TransportDescriptor.h>
+#include <fastrtps/Domain.h>
+
+#include <vector>
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
```
Comment (killight98) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc:17]: include order

**Review by killight98 — CHANGES_REQUESTED** [review]

File: src/lava/magma/runtime/message_infrastructure/CMakeLists.txt (hunk 18 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/CMakeLists.txt:100]: I'll add an exception for that in cmake.

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc (hunk 28 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc:17]: From lint "Should be: fast_dds.h, c system, c++ system"  -> *.h <vector> *.hpp
difficult to be: <fastdds*> <vector>
add NOLINT?

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc (hunk 25 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc:138]: After publisher exist, the subscriber cannot recv the data. I can check the qos config for publisher stop and subscriber still recv...

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.h (hunk 26 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.h:121]: pre-design left... will remove it.

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.h (hunk 27 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.h:25]: from A->Z

File: src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc (hunk 10 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/test/cpptest/test_ddschannel.cc]: ok

File: src/lava/magma/runtime/message_infrastructure/test/test_channel.py (hunk 11 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/test/test_channel.py]: sure

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/core/channel_factory.h (hunk 12 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/core/channel_factory.h]: yes, and I will also change GetDefDDSChannel to GetDDSChannel cause we have no def dds

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc (hunk 24 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc]: yes, I'll remove it.

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc (hunk 15 above)
Comment (hongdami) [reply src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/channel/dds/fast_dds.cc]: miss white space

**Pull request merged**